mcan/src/lib.rs

```diff
@@ -1,6 +1,263 @@
 #![no_std]
-// TODO: Enable when documentation is finished
-// #![warn(missing_docs)]
+#![warn(missing_docs)]
+//! # MCAN
+//!
+//! ## Overview
+//! This crate provides a platform-agnostic CAN HAL.
+//!
+//! It provides the following features:
+//!
+//! - classical CAN and CAN FD with bitrate switching support
+//! - safe Message RAM layouting system via [`SharedMemory`] that prevents
+//!   misconfiguration through compile-time checks
+//! - modular interrupt handling abstractions that enable lock-less usage of HW
+//!   interrupt lines
+//! - message transmission using dedicated buffers, FIFO and priority queue
+//! - message transmission cancellation
+//! - message reception using dedicated buffers and two FIFOs
+//! - filter settings
+//!
+//! MCAN is embedded in the MCU like all other peripherals. The interface
+//! between them includes two clock signal lines, two HW interrupt lines, a
+//! single memory-mapped HW register and a dedicated, shared RAM memory region
+//! (referred to as Message RAM) that both CPU and MCAN can access and share
+//! information through.
+//!
+//! For the MCAN abstractions to be considered operational, this interface has
+//! to be properly configured. The latter is assured through the safety
+//! requirements of [`mcan_core`] traits which platform-specific HALs are
+//! expected to implement.
+//!
+//! In order to use MCAN, one has to instantiate [`CanConfigurable`] and
+//! [`finalize`] it. Its constructor requires an instance of an
+//! [`Dependencies`] implementing struct and holds onto it until it's
+//! [`freed`]. Safety requirements of the `Dependencies` trait
+//! guarantee a correct state of MCAN interfaces during its operation.
+//!
+//! TODO: We do not have `CanConfigurable::free`??
```

Comment:
I guess we'll just have to add it. Bikeshed the name in #28.

Reply:
Done.